drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -336,6 +336,58 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    # All instances for Occurrence Point forms, including inactive/retired, so we can check for legacy data.
+    # Draft forms should never have legacy data, but let's include them just in case.
+    base_scope = Hmis::Form::Instance.with_role(:OCCURRENCE_POINT)
+
+    # All possible form identifiers used for Occurrence Point collection
+    occurrence_point_identifiers = base_scope.pluck(:definition_identifier).uniq
+
+    # Get the latest version of each form definition so we can check for legacy data
+    definitions = Hmis::Form::Definition.where(identifier: occurrence_point_identifiers).latest_versions
+
+    definitions.map do |definition|
+      # Choose the most specific instance for each definition identifier
+      instance_scope = base_scope.where(definition_identifier: definition.identifier).order(updated_at: :desc)
+      best_instance = instance_scope.active.published.detect_best_instance_for_enrollment(enrollment: self)
+
+      # If we didn't find a best_instance, check for legacy data
+      has_any_data = false
+      unless best_instance
+        definition_to_walk = definition.published_version || definition
+
+        definition_to_walk.walk_definition_nodes(as_open_struct: true) do |item|
+          next unless item.mapping.present?
+
+          record_type = item.mapping&.record_type
+          field_name = item.mapping&.field_name&.underscore
+          custom_field_key = item.mapping&.custom_field_key
+
+          next unless record_type == 'ENROLLMENT' || custom_field_key
+
+          if record_type && field_name
+            has_any_data = respond_to?(field_name) && send(field_name).present?
+          elsif custom_field_key
+            # For simplicity, for now, just look for CDEDs where the owner is an Enrollment
+            owner_class_name = self.class.sti_name
+            cded = Hmis::Hud::CustomDataElementDefinition.for_type(owner_class_name).find_by(key: custom_field_key)
+            has_any_data = cded && Hmis::Hud::CustomDataElement.where(data_element_definition: cded, owner: self).any?
+          end
+          break if has_any_data # if we found data, we don't need to keep iterating through the items
+        end
+      end
+
+      # If we found legacy data, then return an instance so we can display it. Prioritize active/published
+      if has_any_data
+        best_instance ||= instance_scope.active.published.first # prioritize active/published instances
+        best_instance ||= instance_scope.first # but if none exist, just return the latest instance
+      end
+
+      best_instance
```

Comment:
This point is moot if we go with an OpenStruct since we won't need to re-load the definition https://github.com/greenriver/hmis-warehouse/pull/4877/files#r1842203473